cfme/fixtures/v2v.py

```diff
@@ -19,19 +23,38 @@ def v2v_providers(request, second_provider, provider):
 
 @pytest.fixture(scope='function')
 def host_creds(request, v2v_providers):
-    """Add credentials to conversation host"""
-    provider = v2v_providers[1]
-    hosts = provider.hosts.all()
+    """Add credentials to VMware and RHV hosts."""
+    if len(v2v_providers) > 2:
+        pytest.skip("There are more than two providers in v2v_providers fixture,"
+                    "which is invalid, skipping.")
     try:
-        hosts = hosts if getattr(request, 'param', '') == 'multi-host' else hosts[0:1]
-        for host in hosts:
-            host_data, = [data for data in provider.data['hosts'] if data['name'] == host.name]
+        vmware_provider, rhv_provider = None, None
+        for provider in v2v_providers:
+            if provider.one_of(VMwareProvider):
+                vmware_provider = provider
+            elif provider.one_of(RHEVMProvider):
+                rhv_provider = provider
+        if not vmware_provider and not rhv_provider:
```

Comment:
What do you think about turning this into an `else` ? 

```
if provider.one_of(VMwareProvider):
    vmware_provider = provider
elif provider.one_of(RHEVMProvider):
￼     rhv_provider = provider
else:
    pytest.skip("Provider {} is not a valid provider for v2v testings.format(provider.name)")
```
Here is what I am thinking: You will have only 2 providers in the v2v_providers list (because of your check above). If one of them does not land in vmware or rhevm provider type, `vmware_provider` or `rhv_provider` is going to be `None`. If this happens on the first iteration of the for loop, then we do not need to bother continuing the second.

Reply:
Yes that may also work. I'll update the PR right away.